src/extension/wps/wps-core/src/main/java/org/geoserver/wps/gs/GetCoveragesValue.java

```diff
@@ -0,0 +1,230 @@
+import org.geotools.geometry.jts.JTS;
+import org.geotools.image.ImageWorker;
+import org.geotools.process.factory.DescribeParameter;
+import org.geotools.process.factory.DescribeProcess;
+import org.geotools.process.factory.DescribeResult;
+import org.geotools.referencing.CRS;
+import org.locationtech.jts.geom.Coordinate;
+
+/** A process that returns values from a coverage at a given location */
+@DescribeProcess(title = "GetCoveragesValue", description = "Returns values from a coverage at a given location")
+public class GetCoveragesValue implements GeoServerProcess {
+    private final Catalog catalog;
+
+    public GetCoveragesValue(Catalog catalog) {
+        this.catalog = catalog;
+    }
+
+    @DescribeResult(name = "result", description = "Output value or values at location", type = ValuesAtPoint.class)
+    public ValuesAtPoint execute(
+            @DescribeParameter(
+                            name = "name",
+                            description =
+                                    "Comma delimited names of rasters, optionally fully qualified (workspace:name), will check for the first one that intersects the point and returns a value")
+                    String name,
+            @DescribeParameter(name = "x", description = "x coordinate of the location to sample") double x,
+            @DescribeParameter(name = "y", description = "y coordinate of the location to sample") double y,
+            @DescribeParameter(
+                            name = "crs",
+                            description =
+                                    "Coordinate Reference System of the x and y coordinates, defaults to first coverage CRS if not set",
+                            min = 0,
+                            max = 1)
+                    String crs)
+            throws IOException {
+        CoverageInfo ci = null;
+        CoordinateReferenceSystem coordinateReferenceSystem = null;
+        String[] names = name.split(",");
+        if (names.length == 0) {
+            throw new WPSException("No coverage names provided");
+        }
+        ci = catalog.getCoverageByName(names[0]);
+        if (ci == null) {
+            throw new WPSException("Could not find coverage " + names[0]);
+        }
+        GridCoverageReader reader = ci.getGridCoverageReader(null, null);
+        GridCoverage2DReader reader2d = (GridCoverage2DReader) reader;
+        GridCoverage2D gridCoverage2D = null;
+        boolean nodata = false;
+        Number[] evaluationValue = null;
+        try {
+            coordinateReferenceSystem =
+                    crs != null && !crs.isEmpty() ? CRS.decode(crs) : reader2d.getCoordinateReferenceSystem();
+        } catch (FactoryException e) {
+            throw new WPSException("Could not decode CRS " + crs, e);
+        }
+        Position position = new Position2D(coordinateReferenceSystem, x, y);
+        if (!coordinateReferenceSystem.equals(reader2d.getCoordinateReferenceSystem())) {
```

Comment:
More robust and less prone to do a transformation when the difference is only some metadata: 

```suggestion
        if (CRS.isTransformationRequired(coordinateReferenceSystem, reader2d.getCoordinateReferenceSystem()) {
```

Reply:
done